ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -37,6 +37,8 @@ let finally = Xapi_stdext_pervasives.Pervasiveext.finally
 
 let domain_shutdown_ack_timeout = ref 60.
 
+let xenopsd_vbd_plug_unplug_legacy = ref true
```

Comment:
New code not active by default, so this should be safe to merge (there are known deadlock bugs with the Serial atoms).

A good way to test might be to set number of workers to 1, it will be slow but it should always be possible to make progress even with 1 (well 2 threads in total, 1 for each queue, the regular and parallel one).